demos/progress/js/progressDemo.js

```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
 
             // after a random delay, do it all over again
-            invokeAfterRandomDelay(function () {
-                simulateTime(percent, steps, stepFunction, finishFunction);
+            demo.timer.invokeAfterRandomDelay(function () {
+                demo.timer.simulateTime(that, stepFunction, finishFunction);
             });
         } else {
             finishFunction();
         }
-
     };
 
-    var timeSimulator = function (percent, steps) {
-        var that = {};
-
-        that.start = function (stepFunction, finishFunction) {
-            // simulate time
-            simulateTime(percent, steps, stepFunction, finishFunction);
-        };
-
-        return that;
-    };
+    fluid.defaults("demo.shoppingDemo", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            submitButton: ".progress-demo-submit button",
+            statusText: ".demoSelector-progress-status-text",
+            restartDemo: ".demoSelector-progress-restart",
+            liveRegion: ".demoSelector-liveRegion",
+            progress: ".demoSelector-progress-theComponent"
+        },
+        strings: {
+            confirmStatus: "Confirm and submit your order.",
+            orderSubmitted: "Order Submitted. Demo finished.",
+            percentCompleted: "%percent% Complete",
+            progressTitle: "Checking inventory, please wait."
+        },
+        events: {
+            afterOrderSubmitted: null,
+            onSubmit: null
+        },
+        listeners: {
+            "onCreate.setInitialStatus": {
+                "this": "{that}.dom.statusText",
+                method: "text",
+                args: ["{that}.options.strings.confirmStatus"]
+            },
+            "onCreate.setAria": {
+                funcName: "demo.shoppingDemo.setAriaAttr",
+                args: ["{that}.dom.liveRegion", "{that}.dom.submitButton", "{that}.containerID"]
+            },
+            "onCreate.bindSubmitEvents": {
+                "this": "{that}.dom.submitButton",
+                "method": "on",
+                args: ["click", "{that}.events.onSubmit.fire"]
+            },
+            "onSubmit.": {
+                funcName: "demo.shoppingDemo.verifyInventory",
+                args: ["{that}"]
+            },
+            "afterOrderSubmitted.confirm": {
+                funcName: "demo.shoppingDemo.confirmOrder",
+                args: ["{that}.dom.statusText", "{that}.options.strings.orderSubmitted", "{that}.dom.restartDemo"]
+            }
+        },
+        members: {
+            containerID: {
+                expander: {
+                    funcName: "fluid.allocateSimpleId",
+                    args: ["{that}.container"]
+                }
+            }
+        },
+        components: {
+            progress: {
+                type: "fluid.progress",
+                container: "{that}.dom.progress",
+                options: {
+                    speed: 1000,
```

Comment:
Would be nice to have this options configurable at the top component level `demo.shoppingDemo`. Also, can you add a code comment like `animation speed in milliseconds` etc?

Same comment applies to line 138, 143.